src/main/java/com/endava/web/application/domain/service/EmployeeService.java

```diff
@@ -0,0 +1,80 @@
+package com.endava.web.application.domain.service;
+
+import com.endava.web.application.domain.dao.DepartmentRepository;
+import com.endava.web.application.domain.dao.EmployeeRepository;
+import com.endava.web.application.domain.exception.exceptions.DepartmentConstraintsException;
+import com.endava.web.application.domain.exception.exceptions.DepartmentDoesNotExistsException;
+import com.endava.web.application.domain.exception.exceptions.NoSuchDepartmentException;
+import com.endava.web.application.domain.exception.exceptions.NoSuchEmployeeException;
+import com.endava.web.application.domain.model.Employee;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class EmployeeService {
+
+    private final EmployeeRepository employeeRepository;
+    private final DepartmentRepository repositoryDepartment;
+
+    public List<Employee> getAllEmployees() {
+        return employeeRepository.findAll();
+    }
+
+    public Employee getEmployee(int employeeId) {
+        Optional<Employee> employee = employeeRepository.findById(employeeId);
+        return employee.orElseThrow(() -> new NoSuchEmployeeException(
+                "EMPLOYEE WITH ID " + employeeId + " DOES NOT EXISTS"));
+    }
+
+    public Employee saveEmployee(Employee employee) {
+        checkEmployeeNameConstraints(employee);
+        if(!repositoryDepartment.findDepartmentById(employee.getDepartment().getId()).isPresent()){
+            throw new NoSuchDepartmentException("DEPARTMENT WITH ID " + employee.getDepartment().getId() + " DOES NOT EXISTS");
+        }
+        return employeeRepository.save(employee);
+    }
+
+    public Employee updateEmployee(Employee employee) {
+        if(!repositoryDepartment.existsById(employee.getDepartment().getId())){
+           throw new DepartmentDoesNotExistsException("DEPARTMENT WITH THIS ID DOES NOT EXISTS");
+        }
+        checkEmployeeNameConstraints(employee);
+        return employeeRepository.save(employee);
+    }
+
+    private void checkEmployeeNameConstraints(Employee employee){
+        if (employee.getFirstName() == null || employee.getFirstName().isEmpty() || employee.getFirstName().trim().isEmpty()) {
+            throw new DepartmentConstraintsException("EMPLOYEE FIRST NAME CANNOT BE NULL, EMPTY OR BLANK");
+        }
+        if (employee.getLastName() == null || employee.getLastName().isEmpty() || employee.getLastName().trim().isEmpty()) {
```

Comment:
all this conditions in if can be replaced with StringUtils.isBlank

Reply:
Done.